monkey/monkey_island/cc/ui/src/components/ui-components/plugins-marketplace/InstalledPlugins.tsx

```diff
@@ -0,0 +1,186 @@
+import React, {useContext, useState} from 'react';
+import {PluginsContext} from '../../contexts/plugins/PluginsContext';
+import {shallowAdditionOfUniqueValueToArray, shallowRemovalOfUniqueValueFromArray} from '../../../utils/objectUtils';
+import {GridActionsCellItem} from '@mui/x-data-grid';
+import DeleteIcon from '@mui/icons-material/Delete';
+import DownloadingIcon from '@mui/icons-material/Downloading';
+import DownloadDoneIcon from '@mui/icons-material/DownloadDone';
+import UpgradeIcon from '@mui/icons-material/Upgrade';
+import RemoveDoneIcon from '@mui/icons-material/RemoveDone';
+import BasePlugins from './BasePlugins';
+import AuthComponent from '../../AuthComponent';
+
+
+const InstalledPlugins = () => {
```

Comment:
If there's an upgrade in process, we need to disable the uninstall button and vice versa.

Reply:
> If there isn't new version, the upgrade needs to be hidden.

Already handled, the upgrade button is disabled in that case.
<img src="https://github.com/guardicore/monkey/assets/44770317/4a0dca30-45b2-47bb-94d3-c37a68175cae" height=125>

> If there's an upgrade in process, we need to disable the uninstall button and vice versa.

Done
<img src="https://github.com/guardicore/monkey/assets/44770317/4afec1f5-e4e5-4843-a0fd-0e66b3754338" height=125>
<img src="https://github.com/guardicore/monkey/assets/44770317/6298b0db-e763-40b2-918c-8172dc44d40e)" height=122>

Reply:
> Although the lists should refresh, let's not rely on it:
> 
> If the uninstall process finishes successfully, we need to hide the upgrade icon (if there is any)
> If the upgrade process finishes successfully, we need to hide the upgrade icon

How would we do that without lists? That's how were tracking state?

Reply:
Not sure what you mean.
We have the lists living in the context.